plugins/inputs/ldap_response/ldap_response.go

```diff
@@ -0,0 +1,209 @@
+package ldap_response
+
+import (
+	ctls "crypto/tls"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"gopkg.in/ldap.v2"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Ldap struct {
+	Host               string
+	Port               int
+	Ssl                string
+	InsecureSkipVerify bool
+	SslCa              string
+	BindDn             string
+	BindPassword       string
+	SearchBase         string
+	SearchFilter       string
+	SearchAttributes   []string
+}
+
+const sampleConfig string = `
+  host = "localhost"
+  port = 389
+
+  # ldaps, starttls, or no encryption. default is an empty string, disabling all encryption.
+  # note that port will likely need to be changed to 636 for ldaps
+  # valid options: "" | "starttls" | "ldaps"
+  ssl = ""
+
+  # skip peer certificate verification. Default is false.
+  insecure_skip_verify = false
+
+  # Path to PEM-encoded Root certificate to use to verify server certificate
+  ssl_ca = "/etc/ssl/certs.pem"
+
+  # dn/password to bind with. If bind_dn is empty, an anonymous bind is performed.
+  bind_dn = ""
+  bind_password = ""
+
+  # base entry for searches
+  search_base = ""
+
+  # ldap search to perform. defaults to "(objectClass=*)" if unspecified.
+  search_filter = ""
+
+  # the attributes to return as fields. defaults to "objectclass" if unspecified.
+  search_attributes = [
+    "attribute1",
+    "attribute2",
+  ]
+`
+
+var DefaultSearchFilter = "(objectClass=*)"
+var DefaultSearchAttributes = []string{"objectclass"}
+
+func (l *Ldap) SampleConfig() string {
+	return sampleConfig
+}
+
+func (l *Ldap) Description() string {
+	return "LDAP Response Input Plugin"
+}
+
+// return an initialized Ldap
+func NewLdap() *Ldap {
+	return &Ldap{
+		Host: "localhost",
+		Port: 389,
+	}
+}
```

Comment:
Please avoid this function and construct the struct instance directly at the caller's location.